### ISSUE TITLE
macOS error: dlsym(0x8a21b620, sqlite3_crsqlitedarwinaarch_init): symbol not found

### ISSUE THREAD

**Issue by simonw** [vlcn-io/cr-sqlite#338]
I'm trying to load the https://github.com/vlcn-io/cr-sqlite/releases/download/v0.15.0-next.2/crsqlite-darwin-aarch64.dylib on macOS.
```
/opt/homebrew/Cellar/sqlite/3.43.0_1/bin/sqlite3
```
```
SQLite version 3.43.0 2023-08-24 12:36:59
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
sqlite> .load crsqlite-darwin-aarch64
Error: dlsym(0x8a62b620, sqlite3_crsqlitedarwinaarch_init): symbol not found
sqlite> 
```
I've tried a few different `sqlite3` binaries I had lying around and got the same error in all of them. What am I doing wrong?

**Comment by simonw** [comment]
Aha - just spotted this: https://github.com/vlcn-io/cr-sqlite/blob/59f03c8a23cfa361ca9bf95c603fc70442419127/README.md#L156-L158

**Comment by simonw** [comment]
This. worked:

    sqlite> .load crsqlite-darwin-aarch64 sqlite3_crsqlite_init

**Comment by simonw** [comment]
Different problem, trying to follow this example in the README:

https://github.com/vlcn-io/cr-sqlite/blob/59f03c8a23cfa361ca9bf95c603fc70442419127/README.md?plain=1#L85-L140

I get:
```
sqlite> insert into crsql_changes
  ("table", pk, cid, val, col_version, db_version, site_id)
  values
  ('foo', 5, 'b', '''thing''', 5, 5, X'7096E2D505314699A59C95FABA14ABB5');
Runtime error: query aborted (4)
```

**Comment by tantaman** [comment]
Ah I haven’t kept the readme up to date with the latest version. The entire
output of crsql changes should be fed into an insert into crsql changes on
another node.

Not at a laptop now but the python hypothesis tests as well as sync tests
would have up to date example uses.

**Comment by simonw** [comment]
Gotcha, looks like this is the test to follow: https://github.com/vlcn-io/cr-sqlite/blob/main/py/correctness/tests/test_sync.py

**Comment by tantaman** [comment]
Yep. That uses the latest build (what is in main and the v0.15.xxx releases)

**Comment by tantaman** [comment]
@simonw  - I published `v0.15` as the official stable release and updated all the docs on the site to match. Should be a smoother experience from here on.

**Comment by simonw** [comment]
Wrote up some of my notes on trying this out here: https://til.simonwillison.net/sqlite/cr-sqlite-macos
